projects/bms_carrier/src/passive_balance.c

```diff
@@ -0,0 +1,38 @@
+#include "passive_balance.h"
+#include "log.h"
+
+// Callback for soft timer
+static void prv_passive_balance_cb(SoftTimerId timer_id, void *context) {
+  passive_balance(context);
+
+  soft_timer_start_millis(PASSIVE_BALANCE_INTERVAL_MS, prv_passive_balance_cb, context, NULL);
+}
+
+StatusCode passive_balance_init(LtcAfeStorage *storage) {
```

Comment:
This should be the AfeStorage, no?

If you end up moving it to being called in cell_sense then you might just be able to take in the result array and the length of it.